Assets/Scripts/navigation/Navigation.cs

```diff
@@ -0,0 +1,186 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+using UnityEngine.UI;
+using TMPro;
+
+namespace nav {
+	public class Navigation : MonoBehaviour
+	{
+		public GameObject TitleScreen;
+		Transform player;
+		private city cities;
+		[SerializeField]
+		private NavMeshAgent AI;
+		[SerializeField]
+		//private LineRenderer lineRenderer;
+		public TMP_Text text;
+		public Image ImageSlot;
+		private const string ResourcePath = "crew_portraits";
+		private const string DefaultPortrait = "crew_portraits/phoenician_sailor";
+		public GameObject menu;
+		public GameObject Navgater;
+		private bool StartNavigation = false;
+		Vector3 nexPoint;
+		private void Start() {
+			cities = new city();
+			menu.SetActive(false);
+			Navgater.SetActive(false);
+			if (AI == null) {
+				AI = GetComponent<NavMeshAgent>();
+			}
+			//if(lineRenderer == null) {
+			//	lineRenderer = GetComponent<LineRenderer>();
+			//}
+		}
+		private void Update() {
+
+			if(!TitleScreen.activeSelf && StartNavigation) {
+				ShowMenu(StartNavigation);
+			}
+		}
+		public static float CalcAngle(Vector3 from, Vector3 to) {
+			Vector3 delta = from - to;
+			return Mathf.Atan2(delta.x, delta.z) * 180 / Mathf.PI;
+		}
+		public static float EulerNormalize(float angle) {
+			while (angle < -180) angle += 360;
+			while (angle > 180) angle -= 360;
+			return angle;
+		}
+		public void Setdestination(string target, int ID) {
+			if (target == null) {
+				Debug.LogError("Didn't find the city");
+				return;
+			}
+			Vector3 targetlocaion = cities.GetCityLocation(target);
+			if (targetlocaion == null) {
+				Debug.LogError("Unable to find city location");
+				return;
+			}
+			//assign image
+			ImageSlot.sprite = Resources.Load<Sprite>(ResourcePath + "/" + ID) ?? Resources.Load<Sprite>(DefaultPortrait);
+			StartNavigation = true;
+			//find player
+			player = GameObject.Find("playerShip").transform.Find("ShipParent").transform.Find("kyrenia_parent").transform;
+			this.transform.position = player.position;
+			float distance = Vector3.Distance(targetlocaion, player.position);
+			float radius = 2;
+			if (distance >= radius) {
+				string Saildirecation = FindPlayerDirection(player); // find what angle is player faceing
```

Comment:
also, are you able to use just Globals.GameVars.playerShip.transform instead of all the fancy .transform.Find(...) stuff you're doing to find the player position? we should at least not use the kyrenia_parent transform for navigation because that gets disabled when you upgrade to the trireme. ShipParent is only offset slightly in y, so I think you could just use the playerShip transform which you can access through Globals.GameVars like i said. it's just likely to bite us later to be referring to the ship with those hardcoded strings.